### PR TITLE
[CNVS Upgrade] Migrating components: Badges & Status Dots

### PR DESCRIPTION
**Pull request by jfurrow**

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @MatApple

File: src/styles/components/badges/styles.less [hunk 1]
```diff
@@ -16,163 +16,167 @@ components/badges.less
...
 --------------------------------------------------------------------------------
 ----------------------------------------------------------------------------- */
 
-.badge {
+& when (@badges-enabled) {
 
-  color: @body-background-color;
-  font-size: @body-text-font-size;
-  line-height: @body-text-line-height;
-  padding: 0 @base-spacing-unit * 0.25;
```
Comment (ashenden) [inline src/styles/components/badges/styles.less]: Not sure what our position is on this stylistically, but for arithmetic like this, I'd aim to place them in explicit parentheses.

File: src/styles/components/badges/styles.less [hunk 2]
```diff
@@ -16,163 +16,167 @@ components/badges.less
...
 
   }
 
-}
 
 
+  .active {
 
-.active {
+    .badge {
```
Comment (ashenden) [inline src/styles/components/badges/styles.less]: Interesting way to apply an active state.  Seems like this could create conflicts in certain scenarios.  I guess it works for now, but might be worth changing in future updates to be more badge specific and less tied to it's parent component.

File: src/styles/components/badges/variables.less [hunk 3]
```diff
@@ -6,6 +6,6 @@ badges/variables.less
 --------------------------------------------------------------------------------
 ----------------------------------------------------------------------------- */
 
-
-
 @badges-enabled:                                                                true;
```
Comment (ashenden) [inline src/styles/components/badges/variables.less]: While we've decided to avoid the comments/spacing in the styles themselves, I'd still love to see it observed in the variables for consistency.  See `menu-tabbed/variables.less` for an example.

**Comment by ashenden** [comment]
Looks good.  Added a couple comments.

File: src/styles/components/badges/styles.less (hunk 2 above)
Comment (jfurrow) [inline src/styles/components/badges/styles.less]: This is how it was before. I totally agree it's error-prone.

File: src/styles/components/badges/styles.less (hunk 1 above)
Comment (jfurrow) [inline src/styles/components/badges/styles.less]: Personally I disagree. Math's order of operations tell us how this will be evaluated :)

File: src/styles/components/badges/variables.less (hunk 3 above)
Comment (jfurrow) [inline src/styles/components/badges/variables.less]: Done!

**Comment by jfurrow** [comment]
@ashenden comments addressed!

File: src/styles/components/badges/styles.less (hunk 1 above)
Comment (ashenden) [inline src/styles/components/badges/styles.less]: http://www.slate.com/articles/health_and_science/science/2013/03/facebook_math_problem_why_pemdas_doesn_t_always_give_a_clear_answer.single.html

tl;dr use parentheses to avoid any chance of confusion.

**Comment by ashenden** [comment]
🚢

File: src/styles/components/badges/styles.less (hunk 1 above)
Comment (jfurrow) [inline src/styles/components/badges/styles.less]: Alright, I'll concede. BUT can we do this in a separate PR, all at once? That would make me happy.

Comment (ashenden) [inline src/styles/components/badges/styles.less]: #deal
